### ISSUE TITLE
Choppy sound and, after a while sound stops completely

### ISSUE THREAD

**Issue by Szvzs** [Edzelf/ESP32Radio-V2#141]
Hello!
I am trying to assemble this radio with PCM5102 and no display. This is my config:
#define DEC_HELIX 
#define DUMMYTFT

this is from the config file:
pin_spi_sck = 5
pin_i2s_bck = 27                                    
pin_i2s_din = 25                                     
pin_i2s_lck = 26 

and this is from serial output:
--- Terminal on COM5 | 115200 8-N-1
--- Available filters and text transformations: colorize, debug, default, direct, esp32_exception_decoder, hexlify, log2file, nocontrol, printable, send_on_enter, time
--- More details at https://bit.ly/pio-monitor-filters
--- Quit: Ctrl+C | Menu: Ctrl+T | Help: Ctrl+T followed by Ctrl+H
����������������������������������������������������������������0x1 (POWERON_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:2
load:0x3fff0030,len:1184
load:0x40078000,len:13232
load:0x40080400,len:3028
entry 0x400805e4
E (557) esp_core_dump_flash: No core dump &�ѥѥ�� found!
E (558) esp_core_dump_flash: No core dump partition found!
[     8][D][esp32-hal-cpu.c:244] setCpuFrequencyMhz(): PLL: 480 / 2 = 240 Mhz, APB: 80000000 Hz

Starting ESP32-radio running on CPU 1 at 240 MHz.
[  3044][I][main.cpp:2584] setup(): [main] Version Thu, 26 Dec 2023 14:30:00 GMT.  Free memory 110580
[  3046][I][main.cpp:2585] setup(): [main] Display type is DUMMY
[  3142][I][main.cpp:2595] setup(): [main] SPIFFS is okay, space 1016801, used 33383
[  3146][I][main.cpp:2616] setup(): [main] Found partition 'nvs     ' at offset 0x009000 with size    20480
[  3148][I][main.cpp:2616] setup(): [main] Found partition 'otadata ' at offset 0x00E000 with size     8192
[  3157][I][main.cpp:2616] setup(): [main] Found partition 'spiffs  ' at offset 0x2F0000 with size  1114112
[  3170][I][main.cpp:2488] fillkeylist(): [main] Read 49 keys from NVS
[  3180][I][main.cpp:1911] readIOprefs(): [main] pin_ir set to 35
[  3182][I][main.cpp:1911] readIOprefs(): [main] pin_i2s_bck set to 27
[  3186][I][main.cpp:1911] readIOprefs(): [main] pin_i2s_lck set to 26
[  3192][I][main.cpp:1911] readIOprefs(): [main] pin_i2s_din set to 25
[  3198][I][main.cpp:1911] readIOprefs(): [main] pin_spi_sck set to 5
[  3204][I][main.cpp:1911] readIOprefs(): [main] pin_spi_miso set to 19
[  3210][I][main.cpp:1911] readIOprefs(): [main] pin_spi_mosi set to 23
[  3227][I][main.cpp:2657] setup(): [main] GPIO0 is HIGH
[  3238][I][main.cpp:2657] setup(): [main] GPIO2 is HIGH
[  3248][I][main.cpp:2657] setup(): [main] GPIO4 is HIGH
[  3258][I][main.cpp:2657] setup(): [main] GPIO5 is HIGH
[  3268][I][main.cpp:2657] setup(): [main] GPIO12 is HIGH
[  3278][I][main.cpp:2657] setup(): [main] GPIO13 is HIGH
[  3288][I][main.cpp:2657] setup(): [main] GPIO14 is HIGH
[  3298][I][main.cpp:2657] setup(): [main] GPIO15 is HIGH
[  3308][I][main.cpp:2657] setup(): [main] GPIO16 is HIGH
[  3318][I][main.cpp:2657] setup(): [main] GPIO17 is HIGH
[  3328][I][main.cpp:2657] setup(): [main] GPIO18 is HIGH
[  3338][I][main.cpp:2657] setup(): [main] GPIO19 is HIGH
[  3348][I][main.cpp:2657] setup(): [main] GPIO21 is HIGH
[  3358][I][main.cpp:2657] setup(): [main] GPIO22 is HIGH
[  3368][I][main.cpp:2657] setup(): [main] GPIO23 is HIGH
[  3378][I][main.cpp:2657] setup(): [main] GPIO25 is HIGH
[  3388][I][main.cpp:2657] setup(): [main] GPIO26 is HIGH
[  3398][I][main.cpp:2657] setup(): [main] GPIO27 is HIGH
[  3408][I][main.cpp:2657] setup(): [main] GPIO32 is HIGH
[  3418][I][main.cpp:2657] setup(): [main] GPIO33 is HIGH
[  3428][I][main.cpp:2657] setup(): [main] GPIO34 is LOW, probably no PULL-UP
[  3438][I][main.cpp:2657] setup(): [main] GPIO35 is LOW, probably no PULL-UP
[  3448][I][main.cpp:2657] setup(): [main] GPIO39 is LOW, probably no PULL-UP
[  3449][I][main.cpp:1757] readprogbuttons(): [main] gpio_00 will execute uppreset = 1
[  3452][I][main.cpp:1757] readprogbuttons(): [main] gpio_12 will execute upvolume = 2
[  3460][I][main.cpp:1757] readprogbuttons(): [main] gpio_13 will execute downvolume = 2
[  3468][I][main.cpp:1757] readprogbuttons(): [main] gpio_14 will execute stop
[  3475][I][main.cpp:1757] readprogbuttons(): [main] gpio_17 will execute resume
[  3482][I][main.cpp:1757] readprogbuttons(): [main] gpio_34 will execute station = icecast.omroep.nl:80/radio1-bb-mp3
[  3492][I][main.cpp:2669] setup(): [main] Enable pin 35 for IR
[  3498][I][main.cpp:2674] setup(): [main] Start DUMMY display
[  3503][E][main.cpp:2691] setup(): [main] Display not activated
[  3508][I][main.cpp:2280] mk_lsan(): [main] Create list with acceptable WiFi networks
[  3516][I][WiFiMulti.cpp:84] addAP(): [WIFI][APlistAdd] add SSID: TP-Link_98F3
[  3523][I][main.cpp:2310] mk_lsan(): [main] Added TP-Link_98F3 to list of networks
[  3531][I][WiFiMulti.cpp:84] addAP(): [WIFI][APlistAdd] add SSID: TP-Link_98F3
[  3537][I][main.cpp:2310] mk_lsan(): [main] Added TP-Link_98F3 to list of networks
[  3546][I][WiFiMulti.cpp:84] addAP(): [WIFI][APlistAdd] add SSID: SSID2
[  3551][I][main.cpp:2310] mk_lsan(): [main] Added SSID2 to list of networks
[  4074][D][WiFiGeneric.cpp:1035] _eventCallback(): Arduino Event: 0 - WIFI_READY
[  4167][V][WiFiGeneric.cpp:340] _arduino_event_cb(): STA Started
[  4167][D][WiFiGeneric.cpp:1035] _eventCallback(): Arduino Event: 2 - STA_START
[  4667][I][main.cpp:866] listNetworks(): [main] Scan Networks
[ 10682][V][WiFiGeneric.cpp:383] _arduino_event_cb(): SCAN Done: ID: 128, Status: 0, Results: 3
[ 10683][D][WiFiGeneric.cpp:1035] _eventCallback(): Arduino Event: 1 - SCAN_DONE
[ 10687][I][main.cpp:868] listNetworks(): [main] Scan completed
[ 10693][I][main.cpp:876] listNetworks(): [main] Number of available networks: 3
[ 10700][I][main.cpp:894] listNetworks(): [main]  1 - TP-Link_98F3              Signal: -66 dBm, Encryption WPA_WPA2_PSK, Acceptable
[ 10711][I][main.cpp:894] listNetworks(): [main]  2 - disabled                  Signal: -67 dBm, Encryption WPA_WPA2_PSK, Not accepttable
[ 10723][I][main.cpp:894] listNetworks(): [main]  3 - Anna                      Signal: -86 dBm, Encryption WPA2_PSK, Not accepttable
[ 10735][I][main.cpp:898] listNetworks(): [main] End of list
[ 10767][I][main.cpp:967] nextPreset(): [main] nextPreset is 0
[ 10783][I][main.cpp:2719] setup(): [main] Connect to network
[ 10785][V][WiFiGeneric.cpp:343] _arduino_event_cb(): STA Stopped
[ 10787][D][WiFiGeneric.cpp:1035] _eventCallback(): Arduino Event: 3 - STA_STOP
[ 11799][D][WiFiGeneric.cpp:1035] _eventCallback(): Arduino Event: 0 - WIFI_READY
[ 11803][V][WiFiGeneric.cpp:340] _arduino_event_cb(): STA Started
[ 11803][D][WiFiGeneric.cpp:1035] _eventCallback(): Arduino Event: 2 - STA_START
[ 19017][V][WiFiGeneric.cpp:383] _arduino_event_cb(): SCAN Done: ID: 129, Status: 0, Results: 4
[ 19018][D][WiFiGeneric.cpp:1035] _eventCallback(): Arduino Event: 1 - SCAN_DONE
[ 19022][I][WiFiMulti.cpp:114] run(): [WIFI] scan done
[ 19027][I][WiFiMulti.cpp:119] run(): [WIFI] 4 networks found
[ 19032][D][WiFiMulti.cpp:151] run():        0: [1][C8:3A:35:B4:57:60] disabled (-64) *
[ 19040][D][WiFiMulti.cpp:149] run():  --->   1: [5][D8:07:B6:7B:98:F3] TP-Link_98F3 (-68) *
[ 19048][D][WiFiMulti.cpp:151] run():        2: [1][18:E8:29:FB:06:D4] Anna (-89) *
[ 19056][D][WiFiMulti.cpp:151] run():        3: [9][D8:47:32:EF:C3:50] TP-Link-G-DB (-92) *
[ 19064][I][WiFiMulti.cpp:160] run(): [WIFI] Connecting BSSID: D8:07:B6:7B:98:F3 SSID: TP-Link_98F3 Channel: 5 (-68)
[ 19075][V][WiFiGeneric.cpp:97] set_esp_interface_ip(): Configuring Station static IP: 0.0.0.0, MASK: 0.0.0.0, GW: 0.0.0.0
[ 19779][V][WiFiGeneric.cpp:355] _arduino_event_cb(): STA Connected: SSID: TP-Link_98F3, BSSID: d8:07:b6:7b:98:f3, Channel: 5, Auth: WPA_WPA2_PSK
[ 19781][D][WiFiGeneric.cpp:1035] _eventCallback(): Arduino Event: 4 - STA_CONNECTED
[ 20480][V][WiFiGeneric.cpp:369] _arduino_event_cb(): STA Got New IP:192.168.0.103
[ 20481][D][WiFiGeneric.cpp:1035] _eventCallback(): Arduino Event: 7 - STA_GOT_IP
[ 20484][D][WiFiGeneric.cpp:1098] _eventCallback(): STA IP: 192.168.0.103, MASK: 255.255.255.0, GW: 192.168.0.1
[ 20494][I][WiFiMulti.cpp:174] run(): [WIFI] Connecting done.
[ 20500][D][WiFiMulti.cpp:175] run(): [WIFI] SSID: TP-Link_98F3
[ 20505][D][WiFiMulti.cpp:176] run(): [WIFI] IP: 192.168.0.103
[ 20511][D][WiFiMulti.cpp:177] run(): [WIFI] MAC: D8:07:B6:7B:98:F3
[ 20517][D][WiFiMulti.cpp:178] run(): [WIFI] Channel: 5
[ 20522][I][main.cpp:1639] connectwifi(): [main] SSID = TP-Link_98F3
[ 20528][I][main.cpp:1643] connectwifi(): [main] IP = 192.168.0.103
[ 20534][I][main.cpp:2729] setup(): [main] Start web server
[ 20541][I][main.cpp:2739] setup(): [main] Network found. Starting clients
[ 20550][I][main.cpp:2773] setup(): [main] MDNS responder started
[ 21554][I][main.cpp:2818] setup(): [main] Rotary encoder is disabled (-1/-1/-1)
[ 21554][I][main.cpp:4375] gettime(): [main] Sync TOD
[ 24965][I][main.cpp:4388] gettime(): [main] TOD synced
[ 24966][I][main.cpp:4518] playtask(): [main] Starting I2S playtask..
[ 24970][I][main.cpp:4551] playtask(): [main] Output to I2S, pins 27, 26 and 25
[ 26966][I][main.cpp:2836] setup(): [main] Playtask started
[ 27168][I][main.cpp:967] nextPreset(): [main] nextPreset is 6
[ 27169][I][main.cpp:3406] radiofuncs(): [main] Radiofuncs cmd is 1
[ 27169][I][main.cpp:4590] playtask(): [main] Playtask stop song
[ 27170][I][main.cpp:1401] connecttohost(): [main] Connect to host icecast.omroep.nl:80/radio1-bb-mp3
[ 27291][I][main.cpp:2543] onConnect(): [main] Connected to host at 145.58.53.148 on port 80
[ 27385][I][main.cpp:1462] connecttohost(): [main] send GET command
[ 27389][I][helixfuncs.h:34] player_setVolume(): [helixfuncs] Volume set to 72
[ 27540][I][main.cpp:3654] handlebyte_ch(): [main] Switch to HEADER
[ 27541][I][main.cpp:3675] handlebyte_ch(): [main] Headerline: Server: Icecast 2.4.0-kh22
[ 27543][I][main.cpp:3675] handlebyte_ch(): [main] Headerline: Connection: Close
[ 27550][I][main.cpp:3675] handlebyte_ch(): [main] Headerline: Pragma: no-cache
[ 27557][I][main.cpp:3675] handlebyte_ch(): [main] Headerline: Expires: Thu, 19 Nov 1981 08:52:00 GMT
[ 27566][I][main.cpp:3675] handlebyte_ch(): [main] Headerline: Cache-Control: no-store, no-cache, private
[ 27576][I][main.cpp:3675] handlebyte_ch(): [main] Headerline: Content-Type: audio/mpeg
[ 27583][I][main.cpp:3675] handlebyte_ch(): [main] Headerline: icy-br: 192
[ 27590][I][main.cpp:3675] handlebyte_ch(): [main] Headerline: ice-audio-info: samplerate=48000;channels=2;bitrate=192
[ 27600][I][main.cpp:3675] handlebyte_ch(): [main] Headerline: icy-genre: Talk
[ 27607][I][main.cpp:3675] handlebyte_ch(): [main] Headerline: icy-metadata: 1
[ 27614][I][main.cpp:3675] handlebyte_ch(): [main] Headerline: icy-name: NPO Radio1
[ 27622][I][main.cpp:3675] handlebyte_ch(): [main] Headerline: icy-pub: 0
[ 27628][I][main.cpp:3675] handlebyte_ch(): [main] Headerline: icy-url: http://www.radio1.nl
[ 27636][I][main.cpp:3675] handlebyte_ch(): [main] Headerline: icy-metaint: 16000
[ 27643][I][main.cpp:3745] handlebyte_ch(): [main] Switch to DATA, bitrate is 192, metaint is 16000
[ 27652][I][main.cpp:4582] playtask(): [main] Playtask start song
[ 27658][I][helixfuncs.h:63] helixInit(): [helixfuncs] helixInit called for audio/mpeg
[ 27667][I][helixfuncs.h:124] playChunk(): [helixfuncs] Sync found at 0x0000
[ 27673][I][helixfuncs.h:170] playChunk(): [helixfuncs] MP3Decode error -2
[ 27679][I][helixfuncs.h:63] helixInit(): [helixfuncs] helixInit called for audio/mpeg
[ 27687][I][helixfuncs.h:124] playChunk(): [helixfuncs] Sync found at 0x0080
[ 27702][I][helixfuncs.h:178] playChunk(): [helixfuncs] Bitrate     is 192000
[ 27703][I][helixfuncs.h:179] playChunk(): [helixfuncs] Samprate    is 48000
[ 27707][I][helixfuncs.h:180] playChunk(): [helixfuncs] Channels    is 2
[ 27717][I][helixfuncs.h:181] playChunk(): [helixfuncs] Bitpersamp  is 16
[ 27720][I][helixfuncs.h:182] playChunk(): [helixfuncs] Outputsamps is 2304
[ 29213][I][main.cpp:1295] showstreamtitle(): [main] Streamtitle found, 50 bytes
[ 29213][I][main.cpp:1296] showstreamtitle(): [main] StreamTitle='NPO Radio 1 - EenVandaag - AVROTROS';
����������������������������������������������������������������Connect(): [main] Host disconnected

Does anyone have any idea why it disconnects? Did i make the right connections and setup in preferences?
Thanks!

**Comment by Edzelf** [comment]
I cannot see if you made the right connections, but the settings are okay.  Also connect the "SCK" of your PCM5102 to ground.

**Comment by Szvzs** [comment]
These are my connections:
pin_spi_sck = 5GPIO and GND (no diffrence if connected)
pin_i2s_bck = 27GPIO
pin_i2s_din = 25GPIO
pin_i2s_lck = 26GPIO
pcm5102 VIN changed to 5V from 3.3V
and GND to GND.

The only difference i noticed is that now the radio plays without interrupts for about 5-10 seconds after it gets choppy and then it stops completely. If i select another another station it also works a couple seconds, and then the same issue.
I found these messages in the serial output after a while: 
[279936][W][AsyncTCP.cpp:943] _poll(): ack timeout 4
[279941][W][AsyncTCP.cpp:943] _poll(): ack timeout 4
[338505][W][AsyncTCP.cpp:950] _poll(): rx timeout 4
[338507][W][AsyncTCP.cpp:950] _poll(): rx timeout 4
[340510][W][AsyncTCP.cpp:943] _poll(): ack timeout 4

The connections are simple, did i forget something?

**Comment by Szvzs** [comment]
I noticed on the bottom of the PCM there are some connections made with soldering, and the connections are named H1L through H4L. Are these connections made for some settings within the PCM?

**Comment by Edzelf** [comment]
I have tested with your configuratieo, see picture below.  3 signal lines, ground and VCC and SCK hard wired (on the module) to ground.  The connection on the backside of the I2S module are 1L, 2L, 3H and 4L.
But even if you remove the PCM5102, the datastream should not be stopped (ack-timeout).
I think there is a problem with the network.  You may try to increase the buffer size (QSIZ near line 157 in main.cpp).

![I2S](https://github.com/Edzelf/ESP32Radio-V2/assets/18257026/4d2bd9fd-ecc5-4e7f-9dc1-8f4dae90a0ce)

**Comment by Szvzs** [comment]
So the connections on the back of the PCM are good, i tried without the PCM being connected, and it does the same (after a while the serial output says rx timeout). Increased QSIZ from 400 to 500, 600, 2000, with no luck. I am 3 meters from my router, recently i did the project with Bluetft and VS1053 and that worked fine. Also tried with another ESP32 board(what i don't have is another PCM but in this case i don't think that it would help). I really wanted to do this without the VS1053 because of space shortage and connection number.

**Comment by Edzelf** [comment]
And what was the problem?

**Comment by Szvzs** [comment]
So, the update is, it works, i downloaded the code again and did the connections from scratch. But it works with high noise, (higher than with the VS1053), not all stations.  I tried multiple variants : 128kB, 190kB and so on. It sounds as if low sound frequencies (base notes) come out on the speaker as scratching noises.. Tried with two separate sources with same output. For sound filters, i used only 2 audio transformers which worked with good results for the VS variant but not with the PCM.
